app/controllers/concerns/blacklight/bookmarks.rb

```diff
@@ -102,9 +102,19 @@ def create
   # Beware, :id is the Solr document_id, not the actual Bookmark id.
   # idempotent, as DELETE is supposed to be.
   def destroy
-    bookmark = current_or_guest_user.bookmarks.find_by(document_id: params[:id], document_type: blacklight_config.document_model.to_s)
```

Comment:
I'm not sure if I'm reading this correctly. Does this mean that an API call could now potentially destroy an un-related users Bookmark?

Reply:
I see the change below now...